Library/PagedContainer/PagedContainerViewController.swift

```diff
@@ -0,0 +1,138 @@
+import Combine
+import Foundation
+import UIKit
+
+open class PagedContainerViewController: UIViewController {
+  private weak var activeController: UIViewController? = nil
+  private var subscriptions = Set<AnyCancellable>()
+  private let viewModel = PagedContainerViewModel()
+
+  // TODO: Use the correct page control, per the designs.
+  // This may exist already in SortPagerViewController, or we can write one in SwiftUI.
+  private lazy var toggle = UISegmentedControl(
+    frame: .zero,
+    actions: []
+  )
+
+  open override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.view.backgroundColor = .white
+    self.view.addSubview(self.toggle)
+    self.toggle.translatesAutoresizingMaskIntoConstraints = false
+    self.toggle.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor).isActive = true
+    self.toggle.widthAnchor.constraint(equalTo: self.view.widthAnchor).isActive = true
+
+    self.viewModel.pageTitles.receive(on: RunLoop.main)
+      .sink { [weak self] titles in
+        self?.configureSegmentedControl(withTitles: titles)
+      }.store(in: &self.subscriptions)
+
+    self.viewModel.displayChildViewControllerAtIndex.receive(on: RunLoop.main)
+      .sink { [weak self] controller, index in
+        self?.showChildController(controller, atIndex: index)
+      }.store(in: &self.subscriptions)
+  }
+
+  open override var shouldAutomaticallyForwardAppearanceMethods: Bool {
+    return false
+  }
+
+  public override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+
+    if self.toggle.selectedSegmentIndex == UISegmentedControl.noSegment {
+      self.viewModel.didSelectPage(atIndex: 0)
+    } else if let activeController = self.activeController {
+      activeController.beginAppearanceTransition(true, animated: animated)
```

Comment:
OK, so all this elaborate `appearanceTransition` stuff. This exists so that the child view controllers will receive `viewWillAppear` with `animated = true`. The reason we need _that_ is because `ActivitiesViewController` filters out calls that are unanimated. I was going to clean up `ActivitiesViewController`, instead, but it turns out that we implicitly use that `animated` as a filter for whether or not the `ActivitiesViewController` is being instantiated by a screenshot test; basically, it was rabbithole, and it seemed clearer to fix it this way.

Reply:
Oh, I think that animated thing is what broke our surveys not showing up in activity, too. So it definitely might be worth looking into at some point but I agree that it's complicated and maybe not worth doing right now (and I definitely think it's not worth doing if there's a chance that we'll rewrite the view controller to stop basically being a webview).

Reply:
I do also think it might be worth leaving a comment in the code about this; it doesn't need to be as detailed, though!

Reply:
Good idea, I'll just pull in this comment and edit it a bit.